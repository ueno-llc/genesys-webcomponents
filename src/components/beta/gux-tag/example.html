<gux-tag-beta>Basic</gux-tag-beta>
<gux-tag-beta color="dark-blue">dark blue</gux-tag-beta>
<gux-tag-beta color="blue">blue</gux-tag-beta>
<gux-tag-beta color="purple">purple</gux-tag-beta>
<gux-tag-beta color="teal">teal</gux-tag-beta>
<gux-tag-beta color="light-purple">light purple</gux-tag-beta>
<gux-tag-beta color="dark-pink">dark pink</gux-tag-beta>
<gux-tag-beta color="dark-purple">dark purple</gux-tag-beta>
<gux-tag-beta color="pink">pink</gux-tag-beta>
<gux-tag-beta color="dark-yellow">dark yellow</gux-tag-beta>
<gux-tag-beta color="yellow">yellow</gux-tag-beta>