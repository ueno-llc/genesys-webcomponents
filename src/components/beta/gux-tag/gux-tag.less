@import (reference) '../../../style/color.less';
// Variables part
@border-radius: 2px;

// Style
gux-tag-beta {
  margin-right: 8px;
  display: inline-block;
  div.gux-tag {
    position: relative;
    margin: 4px 0;
    display: flex;
    margin: 4px 0;

    .gux-tag-text,
      .gux-tag-delete-button {
        background-color: @gux-charcoal-grey;
        height: 22px;
        border-radius: @border-radius;
        text-align: center;
        color: white;
        padding: 0;
      }

      .navy { background-color: @gux-navy; }
      .blue { background-color: @gux-blue; }
      .electric-purple { background-color: @gux-electric-purple; }
      .aqua-green { background-color: @gux-aqua-green; }
      .fuscha { background-color: @gux-fuscha; }
      .dark-purple { background-color: @gux-dark-purple; }
      .bubblegum-pink { background-color: @gux-bubblegum-pink; }
      .olive-green { background-color: @gux-olive-green;}
      .lilac { 
        background-color: @gux-lilac;
        color: @gux-charcoal-grey;
      }
      .yellow {
        background-color: @gux-yellow-green;
        color: @gux-charcoal-grey;
      }

      .gux-tag-text {
        white-space: nowrap;
        padding: 1px 8px;
        line-height: 22px;
      }

      .gux-tag-delete-button {
        width: 0;
        height: 24px;
        display: flex;
        border: none;
        cursor: pointer;
        flex-direction: column;
        justify-content: center;
        border-radius: 0 @border-radius @border-radius 0;
        margin-left: 1px;
      }

      .gux-tag-delete-icon {
        margin: 0 auto;
        display: none;
      }
      .gux-tag-delete-button:active, 
      .gux-tag-delete-button:focus {
        display: flex;
      }

    .show-delete-button {
      width: 24px;
      .gux-tag-delete-icon {
        display: inline-block;
      }
    }
    .active-text-element {
      border-radius: @border-radius 0 0 @border-radius;
    }

    &:hover {
      .gux-tag-text {
        .active-text-element()
      }
      .gux-tag-delete-button {
        .show-delete-button()
      }
    }
    .gux-tag-text:focus {
      .active-text-element()
    }
    .gux-tag-delete-button:focus {
      .show-delete-button()
    }
    .gux-tag-text:focus ~ .gux-tag-delete-button {
      .show-delete-button();
    }
  }
}

// Theming

// Dark
.gux-tag-dark-theme {
  color: @gux-off-white;
}

.gux-dark-theme {
  gux-tag-beta {
    .gux-tag-dark-theme();
  }
}

gux-tag-beta.gux-dark-theme {
  .gux-tag-dark-theme();
}

// Light
.gux-tag-light-theme {
  color: @gux-charcoal-grey;
}

.gux-light-theme {
  gux-tag-beta {
    .gux-tag-light-theme();
  }
}

gux-tag-beta.gux-light-theme {
  .gux-tag-light-theme();
}

// Default Theme
gux-tag-beta {
  .gux-tag-light-theme();
}
