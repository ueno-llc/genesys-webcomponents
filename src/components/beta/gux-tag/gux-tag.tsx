import {
  Component,
  Element,
  h,
  Prop,
  Event,
  EventEmitter
} from '@stencil/core';
import tagResources from './i18n/en.json';
import { buildI18nForComponent } from '../../../i18n';

@Component({
  styleUrl: 'gux-tag.less',
  tag: 'gux-tag-beta'
})
export class GuxTag {
  @Element()
  root: HTMLElement;

  /**
   * Triggered when click on close button
   */
  @Event()
  deleteTag: EventEmitter;

  /**
   * Tag background color.
   */
  @Prop()
  color:
    | 'navy'
    | 'blue'
    | 'electric-purple'
    | 'aqua-green'
    | 'fuscha'
    | 'dark-purple'
    | 'bubblegum-pink'
    | 'olive-green'
    | 'lilac'
    | 'yellow-green';

  /**
   * Index for remove tag
   */
  @Prop()
  tagId: string;

  private i18n: (resourceKey: string, context?: any) => string;

  private handlerClickDeleteTag(): void {
    this.deleteTag.emit(this.tagId);
  }

  private getDeleteButton() {
    return (
      <button
        tabindex="0"
        type="button"
        onClick={this.handlerClickDeleteTag.bind(this)}
        class={`gux-tag-delete-button ${this.color || ''}`}
      >
        <gux-icon
          screenreader-text={this.i18n('delete-tag')}
          icon-name="ic-close"
          class="gux-tag-delete-icon"
        />
      </button>
    );
  }

  async componentWillRender() {
    this.i18n = await buildI18nForComponent(this.root, tagResources);
  }

  render() {
    return (
      <div class="gux-tag">
        <div class={`gux-tag-text ${this.color || ''}`} tabindex="0">
          <slot />
        </div>
        {this.getDeleteButton()}
      </div>
    );
  }
}
