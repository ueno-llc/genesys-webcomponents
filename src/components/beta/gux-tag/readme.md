# gux-tag

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description           | Type                                                                                                                                                   | Default     |
| -------- | --------- | --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------- |
| `color`  | `color`   | Tag background color. | `"aqua-green" \| "blue" \| "bubblegum-pink" \| "dark-purple" \| "electric-purple" \| "fuscha" \| "lilac" \| "navy" \| "olive-green" \| "yellow-green"` | `undefined` |
| `tagId`  | `tag-id`  | Index for remove tag  | `string`                                                                                                                                               | `undefined` |


## Events

| Event       | Description                          | Type               |
| ----------- | ------------------------------------ | ------------------ |
| `deleteTag` | Triggered when click on close button | `CustomEvent<any>` |


## Dependencies

### Used by

 - [gux-tag-popover-beta](../gux-tag-popover)

### Depends on

- [gux-icon](../../stable/gux-icon)

### Graph
```mermaid
graph TD;
  gux-tag-beta --> gux-icon
  gux-tag-popover-beta --> gux-tag-beta
  style gux-tag-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
