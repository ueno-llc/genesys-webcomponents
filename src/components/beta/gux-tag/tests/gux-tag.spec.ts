import { newSpecPage } from '@stencil/core/testing';
import { GuxTag } from '../gux-tag';

describe('gux-tag-beta', () => {
  it('should render', async () => {
    const page = await newSpecPage({
      components: [GuxTag],
      html: `
        <gux-tag-beta></gux-tag-beta>
      `,
      language: 'en'
    });

    expect(page.root).toMatchSnapshot();
  });
});
