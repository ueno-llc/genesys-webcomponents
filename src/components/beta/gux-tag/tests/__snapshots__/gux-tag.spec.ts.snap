// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-tag-beta should render 1`] = `
<gux-tag-beta>
  <!---->
  <div class="gux-tag">
    <div class="gux-tag-text" tabindex="0"></div>
    <button class="gux-tag-delete-button" tabindex="0" type="button">
      <gux-icon class="gux-tag-delete-icon" icon-name="ic-close" screenreader-text="Delete tag"></gux-icon>
    </button>
  </div>
</gux-tag-beta>
`;
